cmd/soroban-rpc/internal/db/event_test.go

```diff
@@ -0,0 +1,167 @@
+							HostFunction: xdr.HostFunction{
+								Type: xdr.HostFunctionTypeHostFunctionTypeInvokeContract,
+								InvokeContract: &xdr.InvokeContractArgs{
+									ContractAddress: xdr.ScAddress{
+										Type:       xdr.ScAddressTypeScAddressTypeContract,
+										ContractId: &xdr.Hash{0x1, 0x2},
+									},
+									FunctionName: "foo",
+									Args:         nil,
+								},
+							},
+							Auth: []xdr.SorobanAuthorizationEntry{},
+						},
+					},
+				})
+		}
+		envelope := xdr.TransactionEnvelope{
+			Type: xdr.EnvelopeTypeEnvelopeTypeTx,
+			V1: &xdr.TransactionV1Envelope{
+				Tx: xdr.Transaction{
+					SourceAccount: xdr.MustMuxedAddress(keypair.MustRandom().Address()),
+					Operations:    operations,
+				},
+			},
+		}
+		txHash, err := network.HashTransactionInEnvelope(envelope, network.FutureNetworkPassphrase)
+		if err != nil {
+			panic(err)
+		}
+
+		txProcessing = append(txProcessing, xdr.TransactionResultMeta{
+			TxApplyProcessing: item,
+			Result: xdr.TransactionResultPair{
+				TransactionHash: txHash,
+			},
+		})
+		components := []xdr.TxSetComponent{
+			{
+				Type: xdr.TxSetComponentTypeTxsetCompTxsMaybeDiscountedFee,
+				TxsMaybeDiscountedFee: &xdr.TxSetComponentTxsMaybeDiscountedFee{
+					Txs: []xdr.TransactionEnvelope{
+						envelope,
+					},
+				},
+			},
+		}
+		phases = append(phases, xdr.TransactionPhase{
+			V:            0,
+			V0Components: &components,
+		})
+	}
+
+	return xdr.LedgerCloseMeta{
+		V: 1,
+		V1: &xdr.LedgerCloseMetaV1{
+			LedgerHeader: xdr.LedgerHeaderHistoryEntry{
+				Hash: xdr.Hash{},
+				Header: xdr.LedgerHeader{
+					ScpValue: xdr.StellarValue{
+						CloseTime: xdr.TimePoint(closeTimestamp),
+					},
+					LedgerSeq: xdr.Uint32(sequence),
+				},
+			},
+			TxSet: xdr.GeneralizedTransactionSet{
+				V: 1,
+				V1TxSet: &xdr.TransactionSetV1{
+					PreviousLedgerHash: xdr.Hash{},
+					Phases:             phases,
+				},
+			},
+			TxProcessing: txProcessing,
+		},
+	}
+}
+
+func TestInsertEvents(t *testing.T) {
+	db := NewTestDB(t)
+	ctx := context.TODO()
+	log := log.DefaultLogger
+	log.SetLevel(logrus.TraceLevel)
+	now := time.Now().UTC()
+
+	writer := NewReadWriter(log, db, interfaces.MakeNoOpDeamon(), 10, 10, passphrase)
+	write, err := writer.NewTx(ctx)
+	require.NoError(t, err)
+	contractID := xdr.Hash([32]byte{})
+	counter := xdr.ScSymbol("COUNTER")
+
+	var txMeta []xdr.TransactionMeta
+	for i := 0; i < 10; i++ {
+		txMeta = append(txMeta, transactionMetaWithEvents(
+			contractEvent(
+				contractID,
+				xdr.ScVec{xdr.ScVal{
+					Type: xdr.ScValTypeScvSymbol,
+					Sym:  &counter,
+				}},
+				xdr.ScVal{
+					Type: xdr.ScValTypeScvSymbol,
+					Sym:  &counter,
+				},
+			),
+		))
+	}
+	ledgerCloseMeta := ledgerCloseMetaWithEvents(1, now.Unix(), txMeta...)
+
+	eventW := write.EventWriter()
+	err = eventW.InsertEvents(ledgerCloseMeta)
+	assert.NoError(t, err)
+
+	//TODO: Call getEvents and validate events data.
```

Comment:
Can you please resolve this before merging?

Reply:
yes, actually ignore this TODO as `getEvents` is backed by memory in this diff, this will be resolved in this follow up PR https://github.com/stellar/soroban-rpc/pull/215 (WIP) @2opremio